account_invoice_tax_required/models/account_invoice.py

```diff
@@ -27,20 +14,22 @@ class AccountInvoice(models.Model):
     def test_invoice_line_tax(self):
         errors = []
         error_template = _("Invoice has a line with product %s with no taxes")
-        for invoice in self:
```

Comment:
Since `test_invoice_line_text` is not intended to be exposed over xml/json-rpc it's good practice to prefix it with underscore: it generally helps avoid security issues.